### PR TITLE
8636 Crash in outliner treeview when closing nested prefabs with shortcut keys.

### PR DESCRIPTION
**Pull request by sphrose**
The problem is happening because the code that is trying to move the focus to the parent in response to the treeview collapsing message is trying to close the previously focused prefab which causes a crash as Qt is already doing that. The fix is to not attempt the close, but only in the case of the left arrow button being used. All other methods of closing the prefab still close the child.

Signed-off-by: sphrose <82213493+sphrose@users.noreply.github.com>

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp [hunk 1]
```diff
@@ -149,12 +150,30 @@ namespace AzToolsFramework::Prefab
...
 
+    PrefabFocusOperationResult PrefabFocusHandler::FocusOnParentOfFocusedPrefab([[maybe_unused]] AzFramework::EntityContextId entityContextId)
+    {
+        return FocusOnParentOfFocusedPrefab(entityContextId, FocusChangeBehavior::CloseCurrentlyFocusedItems);
+    }
+
+    PrefabFocusOperationResult PrefabFocusHandler::FocusOnParentOfFocusedPrefabWithoutClosingCurrent(AzFramework::EntityContextId entityContextId)
+    {
+        return FocusOnParentOfFocusedPrefab(entityContextId, FocusChangeBehavior::IgnoreCurrentlyFocusedItems);
+    }
```
Comment (AMZN-daimini) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp:175]: I'm confused on why we need these helper functions, since both the caller and the underlying `FocusOnParentOfFocusedPrefab` both use the `FocusChangeBehavior` enum to track the behavior.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp [hunk 2]
```diff
@@ -186,7 +205,7 @@ namespace AzToolsFramework::Prefab
         return FocusOnOwningPrefab(focusedInstance->get().GetContainerEntityId());
     }
 
-    PrefabFocusOperationResult PrefabFocusHandler::FocusOnPrefabInstanceOwningEntityId(AZ::EntityId entityId)
+    bool PrefabFocusHandler::FindFocusedInstance(AZ::EntityId entityId, InstanceOptionalReference& outFocusedInstance)
```
Comment (AMZN-daimini) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp:208]: Why the change from PrefabFocusOperationResult to bool with an out argument?
If you need the Success case to return an InstanceOptionalReference, it may be easier to just change the outcome

```
using PrefabFocusOperationResult = AZ::Outcome<InstanceOptionalReference, AZStd::string>;
```

**Review by AMZN-daimini — COMMENTED** [review]
Thanks for taking on this! I understand the logic, but I feel like the implementation could be streamlined (see comments below)

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.h [hunk 3]
```diff
@@ -74,9 +76,18 @@ namespace AzToolsFramework::Prefab
         void OnPrefabTemplateDirtyFlagUpdated(TemplateId templateId, bool status) override;
         
     private:
-        PrefabFocusOperationResult FocusOnPrefabInstance(InstanceOptionalReference focusedInstance);
+        enum FocusChangeBehavior
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.h:79]: update: We should make this `enum class`

Comment (sphrose) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.h]: Bother. Thanks, fixed.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp (hunk 2 above)
Comment (sphrose) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp]: This code has been removed.

**Review by lsemp3d — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
Change looks good, tagging @jonawals to take a look too
